sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/models/event_data.hpp

```diff
@@ -61,6 +37,25 @@ namespace Azure { namespace Messaging { namespace EventHubs { namespace Models {
     std::map<std::string, Azure::Core::Amqp::Models::AmqpValue> Properties;
 
     EventData() = default;
+
+    /** @brief Construct a new EventData object from an initializer list of bytes
+     *
+     * @param body - Body for the newly created EventData.
+     */
+    EventData(std::initializer_list<uint8_t> const& body) : Body(body) {}
```

Comment:
This is the firs time I am seeing `initializer_list` as an input parameter next to vector. Do we need it? If so, what's our guidance on when to have this overload vs not.

Reply:
This was needed to be able to say `EventData foo{1,3,5,7}`. 

But I think the convenience of being able to initialize event data with an array (as opposed to creating a vector and then initializing from the vector) is worth the effort.

Reply:
And the syntax is used extensively through the AMQP Value construct - for the same reasons.